libbeat/autodiscover/providers/kubernetes/pod.go

```diff
@@ -188,10 +190,18 @@ func (p *pod) GenerateHints(event bus.Event) bus.Event {
 	cname := builder.GetContainerName(container)
 	hints := builder.GenerateHints(annotations, cname, p.config.Prefix)
 	p.logger.Debugf("Generated hints %+v", hints)
+
+	// Fall back to defaults on the namespace if there were no hints on the pods
+	if len(hints) == 0 {
+		if rawAnn, ok := kubeMeta["defaults"]; ok {
```

Comment:
It sounds to me that we should do a mix of namespace hints + pod specific hints? Example:

I may have a setting to do json decode for all pods, but for one specific pod in the namespace I may want to tune how multiline goes, without affecting the JSON parsing?